File: backend/app/services/analysis/daily_stats.py
from typing import List, Dict
from collections import Counter
from app.models.chat import ChatMessage


def message_count_by_day(messages: List[ChatMessage]) -> Dict[str, int]:
    """
    回傳每日訊息數量統計：
    {
        "2024-01-01": 當天訊息數量,
        "2024-01-02": 當天訊息數量,
        ...
    }
    """

    counter = Counter()
    for msg in messages:
        date_str = msg.datetime.date().isoformat()
        counter[date_str] += 1

    return dict(counter)

File: frontend/utils/analysis/dailyStats.js
export function messageCountByDay(messages) {
  const counter = {};

  for (const msg of messages) {
    const date = new Date(msg.datetime);
    const isoDate = date.toISOString().split("T")[0];
    counter[isoDate] = (counter[isoDate] || 0) + 1;
  }

  return counter;
}
